manual/src/tutorials/moduleexamples.etex

```diff
@@ -130,50 +114,54 @@ Signatures are interfaces for structures. A signature specifies
 which components of a structure are accessible from the outside, and
 with which type. It can be used to hide some components of a structure
 (e.g. local function definitions) or export some components with a
-restricted type. For instance, the signature below specifies the three
-priority queue operations "empty", "insert" and "extract", but not the
-auxiliary function "remove_top". Similarly, it makes the "queue" type
+restricted type. For instance, the signature below specifies the
+queue operations "empty", "add", "top" and "pop", but not the
+auxiliary function "make". Similarly, it makes the "queue" type
 abstract (by not providing its actual representation as a concrete type).
+This ensures that users of the "Fifo" module cannot violate data
```

Comment:
```suggestion
This ensures that users of the "FIFO" signature cannot violate data
```

Reply:
In this sentence we're talking about the data structure invariants of the specific `Fifo` module, not about any invariant that any implementation of the `FIFO` signature could have.  I'd rather stick with the original wording.